File: algorithms/search/maximum.py
def max_element_index(values: list) -> int:
    index_max_value = 0
    max_value = values[index_max_value]

    for index, value in enumerate(values):
        if value > max_value:
            index_max_value = index
            max_value = value

    return index_max_value


def main():
    values = [1720, 1721, 979, 366, 299, 675, 1456]
    print(f"Index: {max_element_index(values)}")


def test():
    test_input = [1720, 1721, 979, 366, 299, 675, 1456]
    assert max_element_index(test_input) == 1


if __name__ == "__main__":
    test()
    main()

File: algorithms/search/maximum.ts
const maximumSearch = (values: number[]): number => {
  let indexMaxValue = 0;
  let maxValue = values[0];
  for (let index = 0; index < values.length; index++) {
    if (values[index] > maxValue) {
      indexMaxValue = index;
      maxValue = values[index];
    }
  }
  return indexMaxValue;
};

const result = maximumSearch([1720, 1721, 979, 366, 299, 675, 1456]);
console.log(`Index: ${result}`);
export {};
